ghw/libghw.c

```diff
@@ -683,9 +771,9 @@ ghw_read_type (struct ghw_handler *h)
 	      printf ("\n");
 	    h->types[i] = (union ghw_type *) e;
 	    break;
-	  err_b2:
-	    free (e->lits);
-	    free (e);
+	  err_rt_b2:
+	    if (e->lits != NULL) free (e->lits);
```

Comment:
(There is no need to check for NULL when calling `free`)

Reply:
Indeed, removed.